ride-service/src/main/java/com/modsen/rideservice/service/impl/RideServiceImpl.java

```diff
@@ -116,46 +236,68 @@ public void update(RideRequest request, Long id) {
 
     @Override
     public void delete(Long id) {
-        if(rideRepository.findById(id).isEmpty()) {
+        if (rideRepository.findById(id).isEmpty()) {
             log.error("Ride with id {} was not found", id);
             throw new NotFoundException(id);
         }
         rideRepository.deleteById(id);
         log.info("Delete ride with id {}", id);
 
     }
+
     @Override
     @Transactional(readOnly = true)
     public RidesListResponse getRidesByPassengerId(long passengerId, int page, int size, String orderBy) {
         PageRequest pageRequest = getPageRequest(page, size, orderBy);
-        Page<Ride> ridesPage = rideRepository.findAllByPassengerIdAndStatus(passengerId, Status.FINISHED, pageRequest);
-        List<Ride> retrievedRides = ridesPage.getContent();
-        List<RideResponse> rides =retrievedRides.stream().map(this::toDto).collect(Collectors.toList());
+        Page<Ride> ridesPage = rideRepository.findAllByPassengerId(passengerId, pageRequest);
+        List<RideResponse> rides = ridesPage.getContent()
+                .stream().map(ride -> {
+                    RideResponse response = toDto(ride);
+                    checkDriver(response, ride.getDriverId());
+                    checkPassenger(response, passengerId);
+                    return response;
+                }).toList();
         log.info("Retrieving rides for passenger with id {}", passengerId);
         return RidesListResponse.builder()
                 .rides(rides).build();
     }
+
     @Override
     @Transactional(readOnly = true)
     public RidesListResponse getRidesByDriverId(long driverId, int page, int size, String orderBy) {
-
         PageRequest pageRequest = getPageRequest(page, size, orderBy);
-        Page<Ride> ridesPage = rideRepository.findAllByDriverIdAndStatus(driverId, Status.FINISHED, pageRequest);
-        List<Ride> retrievedRides = ridesPage.getContent();
-        List<RideResponse> rides =retrievedRides.stream().map(this::toDto).collect(Collectors.toList());
+        Page<Ride> ridesPage = rideRepository.findAllByDriverId(driverId, pageRequest);
+        List<RideResponse> rides = ridesPage.getContent()
+                .stream().map(ride -> {
+                    RideResponse response = toDto(ride);
+                    checkDriver(response, driverId);
+                    checkPassenger(response, ride.getPassengerId());
+                    return response;
+                }).toList();
         log.info("Retrieving rides for driver with id {}", driverId);
         return RidesListResponse.builder()
                 .rides(rides).build();
     }
+
     @Override
-    public void editStatus(long id, StatusRequest statusRequest){
-        if(rideRepository.findById(id).isEmpty()) {
+    public void editStatus(long id, StatusRequest statusRequest) {
+        if (rideRepository.findById(id).isEmpty()) {
             log.error("Ride with id {} was not found", id);
             throw new NotFoundException(id);
         }
```

Comment:
Так тебе потом не придётся делать ещё один запрос к бд
```suggestion
        Optional<Ride> optionalRide = rideRepository.findById(id);
        if (optionalRide.isEmpty()) {
            log.error("Ride with id {} was not found", id);
            throw new NotFoundException(id);
        }
```